### PR TITLE
fix: Restrict deactivated enterprise user access

### PR DESCRIPTION
**Pull request by suejung-sentry**
Restrict what deactivated Enterprise users can see when the environment has guest access turned off.
- If guest access if off, deactivated users should not be able to see anything
- If guest access is on, deactivated users should be able to see public repos, but not private repos

Closes https://github.com/codecov/engineering-team/issues/1859

### CODE REVIEW

File: graphql_api/types/query/query.py [hunk 1]
```diff
@@ -50,11 +52,17 @@ def resolve_owner(_, info, username):
         if not user or not user.is_authenticated:
             raise UnauthorizedGuestAccess()
 
-    return get_owner(service, username)
+        # per product spec, if guestAccess is off for the environment, the current enterpriseUser
+        # must be "activated" in the given target owner (e.g., "codecov" org) in order to see things
+        target = await get_owner(service, username)
+        if user.ownerid not in target.plan_activated_users:
```
Comment (JerrySentry) [inline graphql_api/types/query/query.py:58]: you can use this helper function: `current_user_part_of_org`

Comment (suejung-sentry) [reply graphql_api/types/query/query.py:58]: Oh nice thanks!
Looking over [that helper](https://github.com/codecov/codecov-api/blob/68ade431abc69558315b571aced42974d5f0f38d/codecov_auth/helpers.py#L27) - Are we diligent about updating an owner (say `suejung-sentry`)'s `organizations` when we adjust the other source of truth on the organization (say `codecov`)'s `plan_activated_users`?
From what I could tell the term `activated` relates to whether the ownerid in the `plan_activated_users` list, and I haven't dug yet into what the owner.organizations is, but I thought it had to do with `accounts`?

Comment (suejung-sentry) [reply graphql_api/types/query/query.py:58]: sounds like the helper may not be equivalent or what's desired here in this chat

Screenshot 2024-10-23 at 8 40 00 AM

File: graphql_api/types/query/query.py [hunk 2]
```diff
@@ -50,11 +52,17 @@ def resolve_owner(_, info, username):
         if not user or not user.is_authenticated:
             raise UnauthorizedGuestAccess()
 
-    return get_owner(service, username)
+        # per product spec, if guestAccess is off for the environment, the current enterpriseUser
```
Comment (ajay-sentry) [inline graphql_api/types/query/query.py:55]: nit: I think the comment states something that can be inferred from the code / PR if needed so not sure if there's much value in having it tbh

**Review by ajay-sentry — APPROVED** [review]
lgtm!

**Comment by suejung-sentry** [comment]
Tested in local different scenarios with Enterprise + guest access. Also smoke-tested in staging that no impact on the cloud deployment. If guest access for the deployment is on, no change to the existing repo-level check of whether the user has permission to see the repo. If guest access is off, must be a planActivatedUser to do anything

**Pull request merged**